### PR TITLE
internal: document that we don't #[ignore] tests

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
r? @flodiebold 

I've noticed that we have some ignored ty tests (one of which actually passes today!). Am I missing a reason why replacing `#[ignore]`s with fixmes would be bad?

File: crates/ide_diagnostics/src/handlers/incorrect_case.rs [hunk 1]
```diff
@@ -341,7 +341,6 @@ mod F {
     }
 
     #[test]
-    #[ignore]
     fn bug_trait_inside_fn() {
```
Comment (matklad) [inline crates/ide_diagnostics/src/handlers/incorrect_case.rs:344]: This one I need to dig into deeper tomorrow -- now I am wholly confused about which wrong behavior is wrong here :D

File: crates/hir_ty/src/tests/traits.rs [hunk 2]
```diff
@@ -1475,7 +1475,6 @@ fn test(
 }
 
 #[test]
-#[ignore]
 fn error_bound_chalk() {
```
Comment (matklad) [inline crates/hir_ty/src/tests/traits.rs:1478]: This one looks like it behaves as I'd expect it to?

File: crates/hir_ty/src/tests/coercion.rs [hunk 3]
```diff
@@ -741,10 +741,25 @@ fn coerce_unsize_trait_object_simple() {
 }
 
 #[test]
-// The rust reference says this should be possible, but rustc doesn't implement
-// it. We used to support it, but Chalk doesn't.
 #[ignore]
 fn coerce_unsize_trait_object_to_trait_object() {
```
Comment (flodiebold) [inline crates/hir_ty/src/tests/coercion.rs:745]: we can probably just remove this?

File: crates/hir_ty/src/tests/traits.rs (hunk 2 above)
Comment (flodiebold) [reply crates/hir_ty/src/tests/traits.rs:1478]: Hm, seems so :shrug: I think we switched to leaving out error bounds completely at some point.

**Review by flodiebold — APPROVED** [review]

**Pull request merged**